google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -372,6 +374,12 @@ public StorageObject create(
     }
   }
 
+  private static String detectContentType(StorageObject object) {
+    return firstNonNull(
```

Comment:
nit : I think you will need to test failed scenario as well.

Reply:
What do you mean?
There are two scenarios: the content type is set and not set. Both cases are covered by tests:

Reply:
what will happen if storage object is null??

Reply:
NullPointerException will be thrown. 
BTW, other methods do not check storage object for null:
              .insert(
                  storageObject.getBucket(), <-- NPE will be thrown here
                  storageObject,
                  new InputStreamContent(detectContentType(storageObject), content));

Reply:
Agreed, It's up to you, It's just suggestion.

Reply:
Thanks